### PR TITLE
add getFeedSuggestions

### PR DESCRIPTION
**Pull request by estrattonbailey**
Adds a `getFeedSuggestions` RPC endpoint that returns a feed generator view for use in the onboarding screens of the app.

### CODE REVIEW

**Review by dholms — APPROVED** [review]
looks good 👍

File: lexicons/app/bsky/actor/getFeedSuggestions.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "lexicon": 1,
+  "id": "app.bsky.actor.getFeedSuggestions",
```
Comment (dholms) [inline lexicons/app/bsky/actor/getFeedSuggestions.json:3]: oh wait I think we want this to be in the `feed` namespace not `actor`

File: lexicons/app/bsky/feed/getFeedSuggestions.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  "lexicon": 1,
+  "id": "app.bsky.feed.getFeedSuggestions",
```
Comment (estrattonbailey) [inline lexicons/app/bsky/feed/getFeedSuggestions.json:3]: Only thing I'm thinking about now is that I like `getSuggestedFeeds` better than the other way around 🤔

Comment (dholms) [reply lexicons/app/bsky/feed/getFeedSuggestions.json:3]: i agree 👍

File: packages/bsky/tests/feed-generation.test.ts [hunk 3]
```diff
@@ -316,6 +328,17 @@ describe('feed generation', () => {
...
   })
 
+  describe('getFeedSuggestions', () => {
+    it('returns list of suggested feed generators', async () => {
+      const resEven = await agent.api.app.bsky.feed.getFeedSuggestions(
+        {},
+        { headers: await network.serviceHeaders(sc.dids.bob) },
+      )
+      expect(forSnapshot(resEven.data)).toMatchSnapshot()
+      expect(resEven.data.feeds.map((fg) => fg.uri)).not.toContain(feedUriPrime) // taken-down
```
Comment (dholms) [inline packages/bsky/tests/feed-generation.test.ts:338]: lol hopefully we're not suggesting any taken down feeds 😅 

but probs good belt & suspenders

File: packages/pds/src/app-view/api/app/bsky/feed/getFeedSuggestions.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { Server } from '../../../../../lexicon'
+import AppContext from '../../../../../context'
+
+export default function (server: Server, ctx: AppContext) {
+  server.app.bsky.feed.getFeedSuggestions({
+    auth: ctx.accessVerifier,
+    handler: async ({ req, auth, params }) => {
+      const requester = auth.credentials.did
+
+      if (await ctx.canProxyRead(req, requester)) {
```
Comment (dholms) [inline packages/pds/src/app-view/api/app/bsky/feed/getFeedSuggestions.ts:10]: sweet yup 👍 

while we still have the appview proxy logic in here, we'll need to toggle the knob that enables proxying this

just remind us to do it when deploying

**Review by dholms — APPROVED** [review]
sweet looks good 👍 

i support the rename that you mentioned

File: packages/pds/src/app-view/api/app/bsky/feed/getFeedSuggestions.ts (hunk 4 above)
Comment (estrattonbailey) [reply packages/pds/src/app-view/api/app/bsky/feed/getFeedSuggestions.ts:10]: Oh, I suppose I can just remove this check entirely, huh

**Pull request merged**